Utilities/StrFmt.cpp

```diff
@@ -152,6 +152,27 @@ void fmt_class_string<std::vector<char>>::format(std::string& out, u64 arg)
 	out.append(obj.cbegin(), obj.cend());
 }
 
+template <>
+void fmt_class_string<std::u8string>::format(std::string& out, u64 arg)
```

Comment:
Concepts can be used for partial specialization as well, u8string, u8string_view, vector<char8_t>, string, string_view and vector<char> can all have one implementation using it.

Reply:
I see what you mean, but it isn't worth it. Concepts are always templates, when full specializations are not. This design prevents instantiation of format functions outside of single cpp file they are specialized in.